### ISSUE TITLE
Stray tag v0.9.1?

### ISSUE THREAD

**Issue by lucc** [qutebrowser/qutebrowser#2284]
In the git history of the master branch there is a commit 73f371a956754c14d35bb2e28a598db5bfb02f91, labeled "Release v0.9.1" and there is also a tag "v0.9.1" at 65852b4b6af694114cea6d34945b8f5d735bcb89. But it seems this tag is no ancestor of master:
```
git merge-base --is-ancestor v0.9.1 master
git merge-base --is-ancestor master v0.9.1
```
both exit with code 1.

Shouldn't the release tag be "in" master?

(Note: I found this when doing `git describe` on master which yields wrong/confusing output because it thinks the last tag was v0.9.0)

**Comment by The-Compiler** [comment]
Patch releases are in release branches, i.e. all `v0.9.x` tags live in the `v0.9.x` branch. This does break a simple `git describe`, but I have no intention of changing that as other branching models have bigger drawbacks IMHO, and changing this now would be a mess either way.

**Comment by lucc** [comment]
Is there any reason that you can not merge the branch `v0.9.x` back into master after each release? Such a merge would probably not push any changes to master (depending on how changes end up in `v0.9.x` in the first place) but would be a kind of "promise" or "statement" that all changes and improvements that are in `v0.9.x` are also in master. If I understand merges correctly that is always the case with a normal merge: On a non technical level the merge commit tells you that the commiter of the merge confirms that he/she has made sure that all relevant changes from both histories are included (especially with confilicts but theoretically with any merge). (On the technical side of things git just provides us with nicely implemented algorithms to help us not to forget any of said changes.)

The added benefit of such a meging of the version branch back into the master branch would be that `git describe` gives more meaningful output and could be used by packagers (see my comment on the [aur/qutebrowser-git](https://aur.archlinux.org/packages/qutebrowser-git/) page).

I just tried the following as a proof of concept:
```sh
git checkout -b poc master
git merge origin/v0.9.x
vi -c DoTheMagic -c wqa # ;)
git commit
```
The result is [here](https://github.com/lucc/qutebrowser/commit/895ced7803b82733258974661241021889b0b120). And `git describe` says `v0.9.1-184-g895ced780` :)

**Comment by The-Compiler** [comment]
I just tried this with the latest master, and it ends up re-adding code which was in v0.9.1 but I removed in master (without a merge-conflict there). I mean, sure, I could make things work by carefully reviewing and editing the merge, but this seems way too error-prone for a small benefit.

> but would be a kind of "promise" or "statement" that all changes and improvements that are in v0.9.x are also in master

I'm also not sure if this is a valid promise in the first place. If a bug ends up being fixed in master by some more complex rewrite (which was needed for other reasons), and in v0.9.x with a simpler fix, it wouldn't hold. As a relatively simple example, look at the refactoring in master (for v0.9.0) in 5463e133eec32452dfd4890da44c853da33c49e2, the much simpler fix in master in 56515321dd3670ea41abed39378cff1b7c984910, and the more complex fix (because hints crashed for all void linux users) on top of v0.8.0 in a43aa52c81ee379031f0d189889e4b45ff94824f.
